### PR TITLE
Empty

### PR DESCRIPTION
**Pull request by etokareva**

### CODE REVIEW

File: src/ng/services/dom/box.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ class BoxCore {
 	}
 
 	cell() {
-		return null;
```
Comment (vkorolev) [inline src/ng/services/dom/box.js]: It doesn't look useful. Do we use it anywhere?

File: src/ng/services/dom/element.js [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
+		return 0;
+	}
+}
 export default class Element {
 	constructor(element) {
 		this.element = element;
 	}
 
+	static get empty() {
+		return new ElementCore();
```
Comment (vkorolev) [inline src/ng/services/dom/element.js:33]: Consider to make it singleton and avoid instantiation on every call

File: src/ng/services/dom/table.spec.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ describe('Table', () => {
 		});
 		it('cell', () => {
 			tableProps.forEach((prop) => {
-				expect(table[prop].cell(0, 0)).to.be.null;
+				expect(table[prop].cell(0, 0)).to.be.an.instanceOf(Object);
```
Comment (vkorolev) [inline src/ng/services/dom/table.spec.js:46]: `to.equal(Cell.empty)` maybe?

File: src/ng/services/dom/element.js (hunk 2 above)
Comment (etokareva) [reply src/ng/services/dom/element.js:33]: Ok, it was attemt to create singleton))) I'll try again.

File: src/ng/services/dom/cell.js [hunk 4]
```diff
@@ -2,16 +2,16 @@ import Element from './element';
 import AppError from 'core/infrastructure/error';
 
 export default class Cell extends Element {
+	static get empty() {
+		return Element.empty;
+	}
```
Comment (klumba12) [inline src/ng/services/dom/cell.js:7]: why element.empty? does element contains all cell properties?

File: src/ng/services/dom/element.js [hunk 5]
```diff
@@ -1,9 +1,42 @@
...
+	constructor() {
+	}
+
+	get model() {
+		return null;
+	}
+
+	rect() {
+		return {};
+	}
```
Comment (klumba12) [inline src/ng/services/dom/element.js:11]: object should have top,left,width,height equal to zero?

**Comment by etokareva** [comment]
Closed as it was pulled to markup-refactoring branch

**Pull request closed without merging**